arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1712,10 +1713,34 @@ int ClusterInfo::ensureIndexCoordinator(
   }
   std::string const idString = arangodb::basics::StringUtils::itoa(iid);
 
+  AgencyComm ac;
   int errorCode;
   try {
-    errorCode = ensureIndexCoordinatorWithoutRollback(
-      databaseName, collectionID, idString, slice, create, compare, resultBuilder, errorMsg, timeout);
+    auto start = std::chrono::steady_clock::now();
+    while (true) { // Keep trying for 2 minutes, if it's preconditions, which are
```

Comment:
I think the code is correct, but I don't understand this code comment. Can you fix it?